packages/client/src/client.js

```diff
@@ -406,10 +406,18 @@ export class PercyClient {
     });
   }
 
-  async uploadComparisonTile(comparisonId, { index = 0, total = 1, filepath, content } = {}) {
+  async uploadComparisonTile(comparisonId, { index = 0, total = 1, filepath, content, sha } = {}) {
     validateId('comparison', comparisonId);
     this.log.debug(`Uploading comparison tile: ${index + 1}/${total} (${comparisonId})...`);
     if (filepath) content = await fs.promises.readFile(filepath);
+    if (sha) {
+      let success = await this.verifyComparisonTile(comparisonId, sha);
+      if (!success) {
+        this.log.error(`Uploading comparison tile for ${sha} failed`);
+        return false;
+      }
+      return true;
+    }
```

Comment:
Ideally would loop in here than on API. API loop will adversely affect API Apdex score

Reply:
looping here would mean multiple API calls, if we loop in API we will have only 1 api call. Also any changes/fixes can be deployed easily there.